experiments/run_slicegpt_perplexity.py

```diff
@@ -233,15 +234,24 @@ def reset_model_device() -> None:
     rotate.rotate_and_slice(model_adapter, train_loader, scheduler, final_orientation=args.final_orientation)
 
     if args.save_dir:
-        path = pathlib.Path(args.save_dir)
-        path.mkdir(parents=True, exist_ok=True)
+        sliced_model_dir = pathlib.Path(args.save_dir)
+        sliced_model_dir.mkdir(parents=True, exist_ok=True)
 
-        model_path = path / f'{pathlib.Path(args.model).name}_{args.sparsity}.pt'
+        sliced_model_name = sliced_model_dir / f'{pathlib.Path(args.model).name}_{args.sparsity}.pt'
 
-        config_path = model_path.with_suffix('.json')
+        config_path = sliced_model_name.with_suffix('.json')
         config_path.write_text(model_adapter.slicing_conf.to_json_string())
 
-        torch.save(model.state_dict(), model_path)
+        # If slicing a local model, also save HF config files in sliced model dir
+        if args.model_path:
+            # copy all config files
+            for file in glob.glob(f"{args.model_path}/*config*.json"):
+                shutil.copy(file, sliced_model_dir)
+            # copy all tokenizer files
+            for file in glob.glob(f"{args.model_path}/*token*.json"):
+                shutil.copy(file, sliced_model_dir)
+
+        torch.save(model.state_dict(), sliced_model_name)
```

Comment:
I would make this the first statement in the save logic, or wrap the above it try-except. So as to avoid the situation when anything goes wrong with write/shutil above and the script terminates without saving the weights that were potentially expensive to obtain.

Reply:
Thanks, good catch.